src/interface/swarm.hpp

```diff
@@ -236,9 +193,17 @@ class Swarm {
   int num_particles_sent_;
   bool finished_transport;
 
+  // Class to store raw pointers to boundary conditions on device. Copy locally for
+  // compute kernel capture.
+  ParticleBoundaries pbounds;
+
```

Comment:
Does this work as intended on a device? IIRC member variables are not captured automatically. Maybe I also haven't fully understood the difference between `pbounds.bounds` and `bounds`.

Reply:
My understanding is member variables are captured automatically (e.g. how `SwarmDeviceContext` puts a bunch of data in a struct that is then all copied onto the device when captured). If this weren't working on device it should fail when the particles regression is run on GPU on the darwin CI right?

`pbounds.bounds` is the array of device pointers to `ParticleBoundary` objects, and `bounds` is the array of `unique_ptr`s that own those pointers and handle automatic deletion at cleanup time. I'll rename `Swarm::bounds` to... um... `bounds_uptrs` and `pbounds` to `bounds_d` (in analogy to `swarm_d`) and `ParticleBoundaries` to `BoundaryDeviceContext`?